src/viewProviders/schemas.ts

```diff
@@ -326,9 +362,40 @@ export class SchemasViewProvider implements vscode.TreeDataProvider<SchemasViewP
     }
   }
 
-  /** Try to reveal this particular schema, if present */
-  revealSchema(schema: Schema): void {
-    this.treeView.reveal(schema, { focus: true, select: true, expand: true });
+  /** Try to reveal+expand schema under a subject, if present */
+  async revealSchema(schemaToShow: Schema): Promise<void> {
+    // Find the subject in the tree view's model.
+    const subject = this.subjectsInTreeView.get(schemaToShow.subject);
+    if (!subject) {
+      logger.error("Strange, couldn't find subject for schema in tree view", {
+        schemaToShow,
+      });
+      return;
+    }
+
+    // The subject may be brand new and have no schemas yet. Expand the subject
+    // to force a fetch of the schemas.
+    if (!subject.schemas) {
+      await this.treeView.reveal(subject, { focus: true, expand: true });
+      // Will now have loaded the schemas for the subject, so now clear to find the schema.
+      // `subject` will be updated with a schemas array.
+    }
+
+    // Find the equivalent schema-within-the-subject by ID. It should have been loaded already, otherwise
+    // something is coded wrong before here.
+    const schema = subject.schemas!.find((schema) => schema.id === schemaToShow.id);
+    if (!schema) {
+      logger.error(
+        `Strange, found subject ${subject.name} but it did not contain the expected schema by id: ${schemaToShow.id}`,
+      );
+      return;
+    }
+
+    try {
+      await this.treeView.reveal(schema, { focus: true, select: true });
+    } catch (e) {
+      logger.error(`Error revealing schema in tree view: ${e instanceof Error ? e.message : e}`);
+    }
```

Comment:
Do we want to send this to Sentry via `logError()`?

Reply:
We could. although hitting it was intermediate code-not-yet-fully-self-consistent fodder, hence growing the catch statement. We oughn't hit it anymore. Will go with Sentry, coming up.